src/main/java/com/lalabrand/ecommerce/user/cart/CartController.java

```diff
@@ -23,10 +26,10 @@ public CartController(CartService cartService, CommonUtils commonUtils) {
 
     @QueryMapping(name = "cart")
     @PreAuthorize("hasAuthority('USER')")
-    public CartDTO findCartByUserId() {
-        Optional<User> user = commonUtils.getCurrentUser();
-        if (user.isPresent()) {
-            return cartService.findCartByUserId(user.get().getId()).orElse(null);
+    public CartDTO findCartForCurrentUser() {
+        UserDetailsImpl user = commonUtils.getCurrentUser();
+        if (user != null) {
+            return cartService.findCartByUserId(user.getId()).orElse(null);
         }
         throw new UsernameNotFoundException("User not found!");
```

Comment:
@denomelchenko user == null is a bad scenario that must be checked. You can assume that the user is not null for a success scenario.

```suggestion
        UserDetailsImpl user = commonUtils.getCurrentUser();
        if (user == null) {
            throw new UsernameNotFoundException("User not found!");
        }
        return cartService.findCartByUserId(user.getId()).orElse(null);
```

Reply:
Fixed)